devicetypes/fibargroup/fibaro-co-sensor-zw5.src/fibaro-co-sensor-zw5.groovy

```diff
@@ -0,0 +1,409 @@
+			state "temperature", label:'${currentValue}°',
+					backgroundColors:[
+							[value: 31, color: "#153591"],
+							[value: 44, color: "#1e9cbb"],
+							[value: 59, color: "#90d2a7"],
+							[value: 74, color: "#44b621"],
+							[value: 84, color: "#f1d801"],
+							[value: 95, color: "#d04e00"],
+							[value: 96, color: "#bc2323"]
+					]
+		}
+
+		valueTile("battery", "device.battery", inactiveLabel: false, width: 2, height: 2, decoration: "flat") {
+			state "battery", label:'${currentValue}%\n battery', unit:"%"
+		}
+
+		standardTile("temperatureAlarm", "device.temperatureAlarm", inactiveLabel: false, width: 3, height: 2, decoration: "flat") {
+			state "cleared", label:'Clear', backgroundColor:"#ffffff" , icon: "https://s3-eu-west-1.amazonaws.com/fibaro-smartthings/coSensor/heat_detector0.png"
+			state "heat", label:'Overheat', backgroundColor:"#d04e00", icon: "https://s3-eu-west-1.amazonaws.com/fibaro-smartthings/coSensor/heat_detector1.png"
+		}
+
+		standardTile("tamper", "device.tamper", inactiveLabel: false, width: 3, height: 2, decoration: "flat") {
+			state "clear", label:'', icon: "https://s3-eu-west-1.amazonaws.com/fibaro-smartthings/coSensor/tamper_detector0.png"
+			state "detected", label:'', icon: "https://s3-eu-west-1.amazonaws.com/fibaro-smartthings/coSensor/tamper_detector100.png"
+		}
+
+		main "FGDW"
+		details(["FGDW","coLevel","temperature","battery","temperatureAlarm","tamper"])
+	}
+
+	preferences {
+
+		input (
+				title: "Fibaro CO Sensor ZW5 manual",
+				description: "Tap to view the manual.",
+				image: "http://manuals.fibaro.com/wp-content/uploads/2017/07/co_icon.png",
+				url: "http://manuals.fibaro.com/content/manuals/en/FGCD-001/FGCD-001-EN-T-v1.1.pdf",
+				type: "href",
+				element: "href"
+		)
+
+		parameterMap().findAll{(it.num as Integer) != 54}.each {
+			input (
+					title: "${it.num}. ${it.title}",
+					description: it.descr,
+					type: "paragraph",
+					element: "paragraph"
+			)
+
+			input (
+					name: it.key,
+					title: null,
+					description: "Default: $it.def" ,
+					type: it.type,
+					options: it.options,
+					range: (it.min != null && it.max != null) ? "${it.min}..${it.max}" : null,
+					defaultValue: it.def,
+					required: false
+			)
+		}
+
+		input ( name: "logging", title: "Logging", type: "boolean", required: false )
+	}
+}
+
+def installed() {
+	sendEvent(name: "checkInterval", value: 86520, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID])
+}
+
+def updated() {
+	if ( state.lastUpdated && (now() - state.lastUpdated) < 500 ) return
+	logging("${device.displayName} - Executing updated()","info")
+
+	if ( (settings.zwaveNotifications as Integer) >= 2 ) {
+		sendEvent(name: "temperatureAlarm", value: "cleared", displayed: false)
+	} else {
+		sendEvent(name: "temperatureAlarm", value: null, displayed: false)
+	}
+
+	syncStart()
+	state.lastUpdated = now()
+}
+
+def ping() {

```

Comment:
If the device is sleepy (sends us a wakeup notification every X hours), it won't respond to these commands.